File: fastapi_server/app/features/review/review_schemas.py
from app.api.api_model import APIModel
from datetime import datetime

class ReviewDetail(APIModel):
    review_id: int
    user_name: str
    product_id: int
    created_at: datetime
    rating: int
    description: str = None

    class from_:
        def Review(o):
            return ReviewDetail(
                review_id = o.id,
                user_name = o.user.name,
                product_id = o.product_id,
                created_at = o.created_at,
                rating = o.rating,
                description = o.description
            )

class ReviewSummary(APIModel):
    rating: int
    description: str = None

# Routes
# GET /products/{product_id}/reviews
# POST /products/{product_id}/reviews
# GET /user/me/reviews
# DELETE /reviews/{review_id}  

class GetProductReviewsResponse(APIModel):
    review_details: list[ReviewDetail]
    average: int
    user_has_review: bool

class PostProductReviewRequest(APIModel):
    review_summary: ReviewSummary

class GetUserReviewsResponse(APIModel):
    review_details: list[ReviewDetail]
    

File: react_client/src/old_features/review/ReviewSchemas.ts


type ReviewDetail = {
    reviewId: number,
    userName: string,
    productId: number,
    createdAt: Date,
    rating: number,
    description?: string | null,
}

type ReviewSummary = {
    rating: number,
    description?: string | null
}

// Routes
// GET /products/{product_id}/reviews
// POST /products/{product_id}/reviews
// GET /user/me/reviews
// DELETE /reviews/{review_id}  

type GetProductReviewsResponse = {
    reviewDetails: ReviewDetail[],
    average: number,
    userHasReview:boolean,
}

type PostProductReviewRequest = {
    reviewSummary: ReviewSummary
}

type GetUserReviewsResponse = {
    reviewDetails: ReviewDetail[],
}

export type {
    ReviewDetail,
    GetProductReviewsResponse,
    PostProductReviewRequest,
    GetUserReviewsResponse,
    ReviewSummary
}
